### PR TITLE
Migrate from isort and flake8 to ruff

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
Blocked by #412 #413. Also need some workflow changes.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,19 +8,14 @@ repos:
...
-        args: [ --max-line-length=120 ]
+      - id: ruff
+        args: ["--fix"]
+        exclude: "tests/input/"
   - repo: https://github.com/psf/black
     rev: 23.10.1
     hooks:
       - id: black
         args: [--safe, --line-length=120]
-  - repo: https://github.com/PyCQA/isort
```
Comment (carlio) [inline .pre-commit-config.yaml:22]: Do ruff + black play well together for sorting imports? I remember that isort came after black due to them conflicting, however that was some time ago and likely fixed.

Comment (Pierre-Sassoulas) [reply .pre-commit-config.yaml:22]: Yeah ruff went out after black so it supports it without config. (there's also ruffmt now, but I like black for having only one option).

**Review by carlio — APPROVED** [review]
Seems okay to me, depends on the other stuff being merged I guess?

**Comment by Pierre-Sassoulas** [comment]
Yeah, I had to rebase on everything to make it work 👌

**Pull request merged**